src/QirRuntime/samples/StandaloneInputReference/qir-driver.cpp

```diff
@@ -0,0 +1,269 @@
+    // Create a QirArray of integer values.
+    QirArray* qirIntegerArray = CreateQirArray(integerVector.data(), integerVector.size());
+
+    // Create a QirArray of double values.
+    QirArray* qirDoubleArray = CreateQirArray(doubleVector.data(), doubleVector.size());
+
+    // Create a QirArray of bool values.
+    bool* boolArray = nullptr;
+    TranslateVectorToBuffer<bool, char>(&boolArray, boolAsCharVector, TranslateCharToBool);
+    QirArray* qirboolArray = CreateQirArray(boolArray, boolAsCharVector.size());
+
+    // Create a QirArray of Pauli values.
+    QirArray* qirPauliArray = CreateQirArray(pauliVector.data(), pauliVector.size());
+
+    // Create a QirRange.
+    QirRange qirRange = TranslateRangeTupleToQirRange(rangeValue);
+
+    // Create a QirArray of Range values.
+    QirRange* rangeArray = nullptr;
+    TranslateVectorToBuffer<QirRange, RangeTuple>(&rangeArray, rangeTupleVector, TranslateRangeTupleToQirRange);
+    QirArray* qirRangeArray = CreateQirArray(rangeArray, rangeTupleVector.size());
+
+    // Create a Result.
+    Result result = TranslateCharToResult(resultAsCharValue);
+
+    // Create a QirArray of Result values.
+    Result* resultArray = nullptr;
+    TranslateVectorToBuffer<Result, char>(
+        &resultArray, resultAsCharVector, TranslateCharToResult);
+
+    QirArray* qirResultArray = CreateQirArray(resultArray, resultAsCharVector.size());
+
+    // Create a QirString.
+    QirString* qirString = quantum__rt__string_create(stringValue.c_str());
+
+    // Redirect the simulator output from std::cout if the --simulator-output option is present.
+    ostream* simulatorOutputStream = &cout;
+    ofstream simulationOutputFileStream;
+    if (!simulationOutputFileOpt->empty())
+    {
+        simulationOutputFileStream.open(simulationOutputFile);
+        Quantum::Qis::Internal::SetOutputStream(simulationOutputFileStream);
+        simulatorOutputStream = &simulationOutputFileStream;
+    }
+
+    // Redirect the Q# operation output from std::cout if the --operation-output option is present.
+    ostream* operationOutputStream = &cout;
+    ofstream operationOutputFileStream;
+    if (!operationOutputFileOpt->empty())
+    {
+        operationOutputFileStream.open(operationOutputFile);
+        operationOutputStream = &operationOutputFileStream;
+    }
+
+    // Run simulation and write the output of the operation to the corresponding stream.
+    int64_t operationOutput = Quantum__StandaloneSupportedInputs__ExerciseInputs__body(
+        intValue, doubleValue, result, qirString, qirRange.start, qirRange.step, qirRange.end);
+
+    simulatorOutputStream->flush();
+    (*operationOutputStream) << operationOutput << endl;
+    operationOutputStream->flush();
+
+    // Close opened file buffers;
+    if (simulationOutputFileStream.is_open())
+    {
+        simulationOutputFileStream.close();
+    }
+
+    if (operationOutputFileStream.is_open())
+    {
+        operationOutputFileStream.close();
+    }
```

Comment:
(Up to you)
Just in case I would `close()` in the reverse order of `open()`. 

Some slightly related info is [here](https://github.com/kuzminrobin/code_review_notes/blob/master/cpp_design_bookmarks.md#know-about-the-compilers-resource-allocation-and-deallocation-order).

Reply:
Changed the ordering.